### DISCUSSION TITLE
Button invert

### DISCUSSION THREAD

**Discussion by benokio** [tttapa/Control-Surface#549] [General]
Hello. I need to send Program changes from 6 buttons. There are leds to each of them. I found an example of using "ManyButtonsSelectorLEDs <N>", but it turned out that my buttons are normally open. In another Issues (* 136) I saw that there is a similar problem and it is recommended to invert the buttons. Can it be done in my case and how? Thanks. This is the code:

ProgramChanger <4> programChanger {
  {
    MIDI_PC :: Acoustic_Grand_Piano, // list of programs
    MIDI_PC :: Rock_Organ,
    MIDI_PC :: Electric_Bass_Pick,
    MIDI_PC :: Bright_Acoustic_Piano,
  },
  CHANNEL_1, // MIDI channel to use
};

ManyButtonsSelectorLEDs <4> programSelector = {
  programChanger, // → selectable
  {{45, 47, 49, 51}}, // → buttonPins
  {{7, 8, 9, 10}}, // → ledPins
};

**Comment by tttapa** [comment]
> my buttons are normally open.

Don't you mean normally closed? If they are NO, they should work out of the box, for NC, you should invert the input.

> Can it be done in my case and how?

This is currently not supported out of the box, but you can easily change the comparison to `HIGH` instead of `LOW` here:

https://github.com/tttapa/Control-Surface/blob/feee931d44d5f20fb7913b171895316e5819549d/src/Selectors/ManyButtonsSelector.hpp#L26-L36

**Reply by benokio** [reply]
Thanks for the fast answer. I tried the change from low to high, but in this code the buttons selecting CC were fixed, but those sending Program Change did not. If it is the opposite (low), only the first two buttons work from both groups. Code:
 
CCSelection <3> cc = {{27, 30, 31}};


ManyButtonsSelectorLEDs <3> selector = {
   cc, // what to select
   {39, 41, 43}, // button pins
   {6, 5, 4}, // led pins
};

ProgramChanger <4> programChanger {
   {
     MIDI_PC :: Acoustic_Grand_Piano, // list of programs
     MIDI_PC :: Rock_Organ,
     MIDI_PC :: Electric_Bass_Pick,
     MIDI_PC :: Bright_Acoustic_Piano,
   },
   CHANNEL_1, // MIDI channel to use
};


ManyButtonsSelectorLEDs <4> programSelector = {
   programChanger, // → selectable
   {{45, 47, 49, 51}}, // → buttonPins
   {{7, 8, 9, 10}}, // → ledPins
};

**Comment by benokio** [comment]
Hello. I found that one of the digital pins was causing me a problem (45). I stopped using it and everything is ok. Thanks a lot for the help.
